### PR TITLE
add support for imagePullSecrets

### PR DESCRIPTION
**Pull request by tobiasmcnulty**

### CODE REVIEW

File: templates/common.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{% if k8s_dockerconfigjson %}
```
Comment (copelco) [inline templates/common.yaml.j2]: The use case where this gets toggled on/off seems slim, but this won't get deleted if you disable ``k8s_dockerconfigjson`` after it was enabled, right?

**Review by copelco — APPROVED** [review]
Looks good to me 👍

File: templates/common.yaml.j2 (hunk 1 above)
Comment (tobiasmcnulty) [reply templates/common.yaml.j2]: Whoops, good catch! I'll see about a ternary for this as well (though yeah, I guess if it persists it's not a huge deal...).

Comment (tobiasmcnulty) [reply templates/common.yaml.j2]: Ternary seems to work, thanks! Going to go ahead and merge.

**Pull request merged**